scalafix-tests/unit/src/test/scala/scalafix/tests/cli/CliGitDiffTests.scala

```diff
@@ -0,0 +1,257 @@
+    val oldCode = "old.scala"
+    val newCode = "new.scala"
+    val newCodeAbsPath = fs.absPath(newCode)
+
+    fs.add(
+      oldCode,
+      """|object OldCode {
+         |  // This is old code, where var's blossom
+         |  var oldVar = 1
+         |}""".stripMargin)
+    git.add(oldCode)
+    addConf(fs, git)
+    git.commit()
+
+    git.checkout("pr-1")
+    fs.replace(
+      oldCode,
+      """|object OldCode {
+         |  // This is old code, where var's blossom
+         |  var oldVar = 1
+         |  // It's not ok to add new vars
+         |  var newVar = 2
+         |}""".stripMargin
+    )
+    fs.mv(oldCode, newCode)
+    git.add(oldCode)
+    git.add(newCode)
+    git.commit()
+
+    val obtained = runDiff(cli)
+
+    val expected =
+      s"""|$newCodeAbsPath:5:3: error: [DisableSyntax.keywords.var] keywords.var is disabled
+          |  var newVar = 2
+          |  ^
+          |""".stripMargin
+
+    println("-------")
+    println(obtained)
+    println("-------")
+    println(expected)
+    println("-------")
+
+    assertNoDiff(obtained, expected)
+  }
+
+  private def runDiff(cli: Cli): String =
+    noColor(cli.run("--diff"))
+
+  private def addConf(fs: Fs, git: Git): Unit = {
+    val confFile = ".scalafix.conf"
+    fs.add(
+      confFile,
+      """|rules = DisableSyntax
+         |DisableSyntax.keywords = [var]""".stripMargin)
+    git.add(confFile)
+  }
+
+  private def noColor(in: String): String =
+    in.replaceAll("\u001B\\[[;\\d]*m", "")
+
+  private def gitTest(name: String)(body: (Fs, Git, Cli) => Unit): Unit = {
+    test(name) {
+      val fs = new Fs()
+      val git = new Git(fs.workingDirectory)
+      val cli = new Cli(fs.workingDirectory)
+
+      body(fs, git, cli)
+    }
+  }
+
+  private class Fs() {
+    val workingDirectory: Path =
+      Files.createTempDirectory("scalafix")
+
+    workingDirectory.toFile.deleteOnExit()
+
+    def add(filename: String, content: String): Unit =
+      write(filename, content, StandardOpenOption.CREATE_NEW)
+
+    def append(filename: String, content: String): Unit =
+      write(filename, content, StandardOpenOption.APPEND)
+
+    def replace(filename: String, content: String): Unit = {
+      rm(filename)
+      add(filename, content)
+    }
+
+    def rm(filename: String): Unit =
+      Files.delete(path(filename))
+
+    def mv(src: String, dst: String): Unit =
+      Files.move(path(src), path(dst))
+
+    def absPath(filename: String): String =
+      path(filename).toAbsolutePath.toString
+
+    private def write(
+        filename: String,
+        content: String,
+        op: StandardOpenOption): Unit = {
+      Files.write(path(filename), content.getBytes, op)
+    }
+
+    private def path(filename: String): Path =
+      workingDirectory.resolve(filename)
+  }
+
+  private class Git(workingDirectory: Path) {
+    import org.eclipse.jgit.api.{Git => JGit}
+
+    private val git = JGit.init().setDirectory(workingDirectory.toFile).call()
```

Comment:
Is it possible to apply a unified diff string? I wonder if we can refactor the tests into a "check" method that takes a name, a StringFS string and expected linter output? Similar to how BaseCliTest supports

Reply:
nope, I don't think there is a good opportunity for refactoring. If we add a lot of stuff in BaseCliTest it's going to be unnecessarily complex.